src/main/java/org/mskcc/oncokb/curation/config/security/SecurityConfigurationAPI.java

```diff
@@ -0,0 +1,103 @@
+import tech.jhipster.config.JHipsterProperties;
+
+/**
+ * We use OAuth2 authentication for the frontend. We also have oncokb-core that needs
+ * to make API requests to this application, so we also need to support JWT token authentication.
+ * In order to use multiple WebSecurityConfigurerAdapter, we need to use requestMatcher() to
+ * let spring know which SecurityConfiguration to use. Our requestMatcher() condition is that any request
+ * with an 'Authorization' header will use SecurityConfigurationAPI, while others will use SecurityConfigurationOAuth.
+ */
+
+@Order(1)
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+@Import(SecurityProblemSupport.class)
+public class SecurityConfigurationAPI extends WebSecurityConfigurerAdapter {
+
+    private final JHipsterProperties jHipsterProperties;
+
+    private final TokenProvider tokenProvider;
+
+    private final CorsFilter corsFilter;
+    private final SecurityProblemSupport problemSupport;
+
+    public SecurityConfigurationAPI(
+        TokenProvider tokenProvider,
+        CorsFilter corsFilter,
+        JHipsterProperties jHipsterProperties,
+        SecurityProblemSupport problemSupport
+    ) {
+        this.tokenProvider = tokenProvider;
+        this.corsFilter = corsFilter;
+        this.problemSupport = problemSupport;
+        this.jHipsterProperties = jHipsterProperties;
+    }
+
+    @Override
+    public void configure(WebSecurity web) {
+        web
+            .ignoring()
+            .antMatchers(HttpMethod.OPTIONS, "/**")
+            .antMatchers("/app/**/*.{js,html}")
+            .antMatchers("/i18n/**")
+            .antMatchers("/content/**")
+            .antMatchers("/swagger-ui/**")
+            .antMatchers("/test/**");
+    }
+
+    @Override
+    public void configure(HttpSecurity http) throws Exception {
+        // @formatter:off
+        http
+            .csrf()
+            .disable()
+            .addFilterBefore(corsFilter, UsernamePasswordAuthenticationFilter.class)
+            .exceptionHandling()
+                .authenticationEntryPoint(problemSupport)
+                .accessDeniedHandler(problemSupport)
+        .and()
+            .headers()
+            .contentSecurityPolicy(jHipsterProperties.getSecurity().getContentSecurityPolicy())
+        .and()
+            .referrerPolicy(ReferrerPolicyHeaderWriter.ReferrerPolicy.STRICT_ORIGIN_WHEN_CROSS_ORIGIN)
+        .and()
+            .permissionsPolicy().policy("camera=(), fullscreen=(self), geolocation=(), gyroscope=(), magnetometer=(), microphone=(), midi=(), payment=(), sync-xhr=()")
+        .and()
+            .frameOptions()
+            .deny()
+        .and()
+            .sessionManagement()
+            .sessionCreationPolicy(SessionCreationPolicy.STATELESS)
+        .and()
+            .requestMatcher(new RequestHeaderRequestMatcher("Authorization"))
+            .authorizeRequests()
+            .antMatchers("/api/**").authenticated()
```

Comment:
We should have this clause at the very top of the configure. Most of other configurations probably not applicable.

Reply:
I moved the condition `.requestMatcher(new RequestHeaderRequestMatcher(JWTFilter.AUTHORIZATION_HEADER))` to the top, so its clear that this config is only for requests with Authorization header.
The `.antMatcher(/api/**) cannot be moved to top because both the JWT and OAuth security config should accept requests to api endpoints.